ocf_data_sampler/torch_datasets/sample/uk_regional.py

```diff
@@ -22,22 +22,167 @@ def __init__(self, data: NumpySample) -> None:
+        if config is None:
+            config = {
+                "required_keys": [
+                    GSPSampleKey.gsp,
+                    NWPSampleKey.nwp,
+                    SatelliteSampleKey.satellite_actual,
+                ],
+            }
+
+        # Check for required keys
+        for key in config.get("required_keys", []):
+            if key not in self._data:
+                raise ValueError(f"Missing required key: {key}")
+
+        expected_shapes = config.get("expected_shapes", {}).copy()
+
+        # Calculate expected shapes from config if available
+        if "input_data" in config:
+            if GSPSampleKey.gsp not in expected_shapes and "gsp" in config["input_data"]:
+                gsp_config = config["input_data"]["gsp"]
+                time_span = (
+                    gsp_config["interval_end_minutes"] -
+                    gsp_config["interval_start_minutes"]
+                )
+                resolution = gsp_config["time_resolution_minutes"]
+                expected_length = (time_span // resolution) + 1
+                expected_shapes[GSPSampleKey.gsp] = (expected_length,)
+
+            # Calculate NWP shape
+            if "nwp" in config["input_data"]:
+                for provider in config["input_data"]["nwp"].values():
+                    if isinstance(provider, dict) and "image_size_pixels_height" in provider:
+                        config["nwp_shape"] = (
+                            provider["image_size_pixels_height"],
+                            provider["image_size_pixels_width"],
+                        )
+                        break
+
+            # Calculate satellite shape
+            if "satellite" in config["input_data"]:
+                sat_config = config["input_data"]["satellite"]
+                has_height = "image_size_pixels_height" in sat_config
+                has_width = "image_size_pixels_width" in sat_config
+                if has_height and has_width:
+                    channels = len(sat_config.get("channels", []))
+                    interval_end = sat_config["interval_end_minutes"]
+                    interval_start = sat_config["interval_start_minutes"]
+                    time_span = interval_end - interval_start
+                    resolution = sat_config["time_resolution_minutes"]
+                    time_steps = (time_span // resolution) + 1
+
+                    satellite_shape = (
+                        time_steps,
+                        channels,
+                        sat_config["image_size_pixels_height"],
+                        sat_config["image_size_pixels_width"],
+                    )
+                    config["satellite_shape"] = satellite_shape
+
+        # Check data shapes against calculated or config given shapes
+        for key, expected_shape in expected_shapes.items():
+            if key in self._data:
+                actual_shape = self._data[key].shape
+
+                # Check dimensions match
```

Comment:
could make check for dimmesionas and check a function, then gets re used in GSP, Satellite and NWP